src/stores/TagOrderStore.js

```diff
@@ -165,13 +175,15 @@ class TagOrderStore extends Store {
     _mergeGroupsAndTags() {
         const groupIds = this._state.joinedGroupIds || [];
         const tags = this._state.orderedTagsAccountData || [];
+        const removedTags = this._state.removedTagsAccountData || [];
```

Comment:
Feels like we may want to turn this into a set (or even store it on the store as a set) to make the membership testing less O(n^2).